### PR TITLE
Add info about the default service name

### PR DESCRIPTION
**Pull request by pjanotti**
## Why

Especially for IIS configuring the `SIGNALFX_SERVICE_NAME` can be cumbersome to configure. We want to avoid this configuration if possible.

## What

Documenting the default service name and making the case that the user just needs to set it if the default doesn't work for their scenario.

## Tests

N/A (docs only)

### CODE REVIEW

File: docs/default-service-name.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+the [`SIGNALFX_SERVICE_NAME`](advanced-config.md) setting.
+The default service name is captured via the steps below,
+stopping at the first one that succeeds:
+
+1. For the [SignalFx .NET Tracing Azure Site Extension](../shared/src/azure-site-extension/README.md)
+the default service name is the __site name__ as defined by the
+[WEBSITE_SITE_NAME](https://docs.microsoft.com/en-us/azure/app-service/reference-app-settings?tabs=kudu%2Cdotnet) 
+environment variable.
+
+2. For an ASP.NET hosted application, eg.: IIS .NET Framework apps, the default service name is `SiteName[/VirtualPath]`.
```
Comment (theletterf) [inline docs/default-service-name.md:13]:
```suggestion
2. For an ASP.NET hosted application, for example IIS .NET Framework apps, the default service name is `SiteName[/VirtualPath]`.
```

File: docs/default-service-name.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+3. For other applications the name of the [entry assembly](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getentryassembly?view=net-6.0)
+is used as the default service name. Typically this is the name of your .NET project file.
+
+4. In the unlikely case that the [entry assembly is not available](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getentryassembly?view=net-6.0#remarks).
```
Comment (theletterf) [inline docs/default-service-name.md:18]:
```suggestion
4. If the [entry assembly is not available](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getentryassembly?view=net-6.0#remarks).
```

File: docs/iis-instrumentation.md [hunk 3]
```diff
@@ -4,7 +4,11 @@ By default, the installer enables IIS instrumentation for .NET Framework
 by setting the `Environment` registry key for W3SVC and WAS services
 located in the `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services` folder.
 
-Configure `SIGNALFX_SERVICE_NAME` as described in [advanced-config.md](advanced-config.md#configuration-methods).
+For ASP.NET applications (ie.: those IIS applications running on the .NET Framework) the
```
Comment (theletterf) [inline docs/iis-instrumentation.md:7]:
```suggestion
For ASP.NET applications, for example IIS applications running on the .NET Framework, the
```

File: docs/iis-instrumentation.md [hunk 4]
```diff
@@ -4,7 +4,11 @@ By default, the installer enables IIS instrumentation for .NET Framework
...
+default service name is `ServiceName[\VirtualPath]`.
+For ASP.NET Core applications the default service name is the entry assembly name, typically
+the name of your .NET Core project.
+If the defaults above don't fit well with your usage or naming conventions configure `SIGNALFX_SERVICE_NAME` as described in [advanced-config.md](advanced-config.md#configuration-methods).
```
Comment (theletterf) [inline docs/iis-instrumentation.md:11]:
```suggestion
If the defaults don't fit well with your usage or naming conventions configure `SIGNALFX_SERVICE_NAME` as described in [advanced-config.md](advanced-config.md#configuration-methods).
```

**Review by theletterf — CHANGES_REQUESTED** [review]
Thanks! Just some nits.

**Comment by pjanotti** [comment]
Thanks for the suggestions @theletterf

**Review by Kielek — APPROVED** [review]

File: docs/iis-instrumentation.md [hunk 5]
```diff
@@ -4,7 +4,11 @@ By default, the installer enables IIS instrumentation for .NET Framework
 by setting the `Environment` registry key for W3SVC and WAS services
 located in the `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services` folder.
 
-Configure `SIGNALFX_SERVICE_NAME` as described in [advanced-config.md](advanced-config.md#configuration-methods).
+For ASP.NET applications, for example IIS applications running on the .NET Framework, the
+default service name is `ServiceName[\VirtualPath]`.
```
Comment (pellared) [inline docs/iis-instrumentation.md:8]: Is it `ServiceName[\VirtualPath]` or `ServiceName[/VirtualPath]` ?

**Review by theletterf — APPROVED** [review]

File: docs/iis-instrumentation.md (hunk 5 above)
Comment (pjanotti) [reply docs/iis-instrumentation.md:8]: The second, will fix it before merging.

**Comment by pjanotti** [comment]
Test run failure:
```terminal
Error: [xUnit.net 00:01:07.03]     Datadog.Trace.ClrProfiler.IntegrationTests.SmokeTests.AssemblyLoadFileNotFoundExceptionSmokeTest.NoExceptions [FAIL]
    Failed Datadog.Trace.ClrProfiler.IntegrationTests.SmokeTests.AssemblyLoadFileNotFoundExceptionSmokeTest.NoExceptions [1 m]
    Error Message:
     System.TimeoutException : The smoke test is running for too long or was lost.
    Stack Trace:
       at Datadog.Trace.ClrProfiler.IntegrationTests.SmokeTests.SmokeTestBase.CheckForSmoke(Boolean shouldDeserializeTraces) in Y:\tracer\test\Datadog.Trace.ClrProfiler.IntegrationTests\SmokeTests\SmokeTestBase.cs:line 127
    Standard Output Messages:
   Application path: Y:\tracer\test/test-applications/regression\AssemblyLoad.FileNotFoundException\bin\X64\Release\net461\AssemblyLoad.FileNotFoundException.exe
   Executable path: Y:\tracer\test/test-applications/regression\AssemblyLoad.FileNotFoundException\bin\X64\Release\net461\AssemblyLoad.FileNotFoundException.exe
   Assigning port 50552 for the aspNetCorePort.
   Assigned port 50553 for the agentPort.
   The smoke test is running for too long or was lost.
   StandardOutput:
   All is well!
   ```

**Pull request merged**